### PR TITLE
WebAppCleanUp Utility

### PR DESCRIPTION
**Pull request by drbruce-git**
Groovy script used to delete DBB web application collections and build groups that are no longer needed

### CODE REVIEW

File: Utilities/WebAppCleanUp/WebAppCleanUp.groovy [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+   		assert properties.pwFile : "Missing 'pwFile' argument/property"
+   	}
+   	
+   	//create repository client
+   	println "Creating repository client for ${properties.url}"
+   	repoClient = new RepositoryClient().forceSSLTrusted(true).url(properties.url).userId(properties.id)
+   	if (properties.pw)
+   	   repoClient.setPassword(properties.pw)
+   	else
+   	   repoClient.setPasswordFile(properties.pwFile)
```
Comment (dennis-behm) [inline Utilities/WebAppCleanUp/WebAppCleanUp.groovy:92]: I don't think that this API exists. The RepositoryClient accepts a java.io.File, but not a String value
```
public void setPasswordFile(java.io.File passwordFile)
```

**Review by dennis-behm — CHANGES_REQUESTED** [review]
Can you please double-check with the pwFile?

File: Utilities/WebAppCleanUp/WebAppCleanUp.groovy [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+// run setup tasks
+setup(args)
+
+// process build group list
+println "** Deleting build groups: $groups"
+groups.each { group ->
+  String name = group.trim()
+  repoClient.deleteBuildResults(name)
+  println "* Deleting build group '$name' -> Status = ${repoClient.getLastStatus()}"
```
Comment (jbyibm) [inline Utilities/WebAppCleanUp/WebAppCleanUp.groovy:20]: In general, we put 3 * in our groovy script for sublevel log.

File: Utilities/WebAppCleanUp/WebAppCleanUp.groovy [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+// process collections list
+println "** Deleting collections: $collections"
+collections.each { collection ->
+  String name = collection.trim()
+  repoClient.deleteCollection(name)
+  println "* Deleting collection '$name' -> Status = ${repoClient.getLastStatus()}"
```
Comment (jbyibm) [inline Utilities/WebAppCleanUp/WebAppCleanUp.groovy:28]: In general, we put 3 * in our groovy script for sublevel log.

**Review by jbyibm — COMMENTED** [review]
I agree with Dennis about the password file API you need to check. I just added a comment about stars in comment sublevel.

**Review by jbyibm — APPROVED** [review]

**Review by dennis-behm — APPROVED** [review]
Many thanks!

**Pull request merged**